app/components/identity-steps/step-three.hbs

```diff
@@ -1,9 +1,74 @@
-    <h3 data-test-getting-started-heading class="heading__h3">Chaincode Generation</h3>
-    <p data-test-getting-started-paragraph-1 class="p2-description">A private that you need to use in your
-        profile service URL and deploy for source
-        that you're the source of the URL
-    </p>
-    <p data-test-getting-started-paragraph-2 class="p2-description">https://github.com/identity-service/instructions.md
-    </p>
-    <Reusables::Button @text="Next" @variant='dark' @onClick={{@startHandler}} @test='identity-next'
-        @type='button' />
+<div data-test='chaincode' class='chaincode-page'>
+  <h3 data-test='heading' class='chaincode-page__heading heading__h3'>
+    Chaincode Generation
+  </h3>
+  <p data-test='description' class='chaincode-page__description'>
+    A private key that you need to use in your profile service URL and deploy
+    for source that you're the source of the URL
+  </p>
+  {{#if @isChaincodeClicked}}
+    <div data-test='chaincode-container' class='chaincode-container'>
+      <div
+        data-test='chaincode-container__value'
+        class={{if
+          this.hideChaincode
+          'chaincode-container__value-invisible'
+          'chaincode-container__value-visible'
+        }}
+      >
+        {{#if this.hideChaincode}}
+          ********************
+        {{else}}
+          {{@chaincode}}
+        {{/if}}
+      </div>
+      <div
+        data-test='chaincode-container__action'
```

Comment:
BEM is only for class naming, you can use normal naming in data-test selectors with `-`